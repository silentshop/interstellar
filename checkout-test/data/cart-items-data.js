// const cartItem = {
//     name: string,
//     price: number,
//     description: string,
//     rating: number
// }

const cartItems = [
    {
        name: "CartItem 1",
        // price: 20,
        fixedUSDAmount: 20,
        acceptedAsset: { balance: 0, asset_type: "alpha-numeric-4", asset_code: "BTC" },
        description: "This is a product.",
        rating: 2
    },
];

module.exports = cartItems;