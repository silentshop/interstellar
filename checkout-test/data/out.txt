[ { code: 'XLM',
    issuer: null,
    domain: 'native',
    price_USD: 0.298234 },
  { code: 'RMT',
    issuer: 'GDEGOXPCHXWFYY234D2YZSPEJ24BX42ESJNVHY5H7TWWQSYRN5ZKZE3N',
    domain: 'sureremit.co',
    price_USD: 0.026006 },
  { code: 'MOBI',
    issuer: 'GA6HCMBLTZS5VYYBCATRBRZ3BZJMAFUDKYYF6AH6MVCMGWMRDNSWJPIH',
    domain: 'mobius.network',
    price_USD: 0.063619 },
  { code: 'CNY',
    issuer: 'GAREELUB43IRHWEASCFBLKHURCGMHE5IF6XSE7EXDLACYHGRHM43RFOX',
    domain: 'ripplefox.com',
    price_USD: 0.14949 },
  { code: 'REPO',
    issuer: 'GCZNF24HPMYTV6NOEHI7Q5RJFFUI23JKUKY3H3XTQAFBQIBOHD5OXG3B',
    domain: 'repocoin.io',
    price_USD: 0.13286 },
  { code: 'KIN',
    issuer: 'GBDEVU63Y6NTHJQQZIKVTC23NWLQVP3WJ2RI2OTSJTNYOIGICST6DUXR',
    domain: 'apay.io',
    price_USD: 0.0001997 },
  { code: 'NGNT',
    issuer: 'GAWODAROMJ33V5YDFY3NPYTHVYQG7MJXVJ2ND3AOGIHYRWINES6ACCPD',
    domain: 'cowrie.exchange',
    price_USD: 0.0028269 },
  { code: 'SLT',
    issuer: 'GCKA6K5PCQ6PNF5RQBF7PQDJWRHO6UOGFMRLK3DYHDOI244V47XKQ4GP',
    domain: 'smartlands.io',
    price_USD: 1.9599 },
  { code: 'BTC',
    issuer: 'GATEMHCCKCY67ZUCKTROYN24ZYT5GK4EQZ65JJLDHKHRUZI3EUEKMTCH',
    domain: 'naobtc.com',
    price_USD: 7550.23 },
  { code: 'ETH',
    issuer: 'GBSTRH4QOTWNSVA6E4HFERETX4ZLSR3CIUBLK7AXYII277PFJC4BBYOG',
    domain: 'stronghold.co',
    price_USD: 466.19 },
  { code: 'BTC',
    issuer: 'GAUTUYY2THLF7SGITDFMXJVYH3LHDSMGEAKSBU267M2K7A3W543CKUEF',
    domain: 'apay.io',
    price_USD: 7647.03 },
  { code: 'ETH',
    issuer: 'GBDEVU63Y6NTHJQQZIKVTC23NWLQVP3WJ2RI2OTSJTNYOIGICST6DUXR',
    domain: 'apay.io',
    price_USD: 469.01 },
  { code: 'LTC',
    issuer: 'GC5LOR3BK6KIOK7GKAUD5EGHQCMFOGHJTC7I3ELB66PTDFXORC2VM5LP',
    domain: 'apay.io',
    price_USD: 85.978 },
  { code: 'TARI',
    issuer: 'GD7UVDDJHJYKUXB4SJFIC6VJDQ4YADQCMRN3KLHJFV4H6NIUAEREVCO7',
    domain: 'cryptotari.io',
    price_USD: 0.20913 },
  { code: 'XRP',
    issuer: 'GA7FCCMTTSUIC37PODEL6EOOSPDRILP6OQI5FWCWDDVDBLJV72W6RINZ',
    domain: 'vcbear.net',
    price_USD: 0.62518 },
  { code: 'BCH',
    issuer: 'GAEGOS7I6TYZSVHPSN76RSPIVL3SELATXZWLFO4DIAFYJF7MPAHFE7H4',
    domain: 'apay.io',
    price_USD: 825.81 } ]
