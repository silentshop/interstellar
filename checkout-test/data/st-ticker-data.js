const ST_TickerIDs = {
    XLM: { 
        code: 'XLM', 
        issuer: null,
        domain: 'native'
    },
    RMT: { 
        code: 'RMT', 
        issuer: 'GDEGOXPCHXWFYY234D2YZSPEJ24BX42ESJNVHY5H7TWWQSYRN5ZKZE3N',
        domain: 'sureremit.co'
    },
    MOBI: {
        code: 'MOBI', 
        issuer: 'GA6HCMBLTZS5VYYBCATRBRZ3BZJMAFUDKYYF6AH6MVCMGWMRDNSWJPIH',
        domain: 'mobius.network'
    },
    REPO: {
        code: 'REPO', 
        issuer: 'GCZNF24HPMYTV6NOEHI7Q5RJFFUI23JKUKY3H3XTQAFBQIBOHD5OXG3B',
        domain: 'repocoin.io' 
    },
    CNY: {
        code: 'CNY', 
        issuer: 'GAREELUB43IRHWEASCFBLKHURCGMHE5IF6XSE7EXDLACYHGRHM43RFOX',
        domain: 'ripplefox.com'
    },
    SLT: {
        code: 'SLT', 
        issuer: 'GCKA6K5PCQ6PNF5RQBF7PQDJWRHO6UOGFMRLK3DYHDOI244V47XKQ4GP',
        domain: 'smartlands.io'
    },
    NGNT: {
        code: 'NGNT', 
        issuer: 'GAWODAROMJ33V5YDFY3NPYTHVYQG7MJXVJ2ND3AOGIHYRWINES6ACCPD',
        domain: 'cowrie.exchange'
    },
    // ETH: {
        ETH_GBSTRH4QOTWNSVA6E4HFERETX4ZLSR3CIUBLK7AXYII277PFJC4BBYOG: {
            code: 'ETH', 
            issuer: 'GBSTRH4QOTWNSVA6E4HFERETX4ZLSR3CIUBLK7AXYII277PFJC4BBYOG',
            domain: 'stronghold.co'
        },
        ETH_GBDEVU63Y6NTHJQQZIKVTC23NWLQVP3WJ2RI2OTSJTNYOIGICST6DUXR: {
            code: 'ETH', 
            issuer: 'GBDEVU63Y6NTHJQQZIKVTC23NWLQVP3WJ2RI2OTSJTNYOIGICST6DUXR',
            domain: 'apay.io'
        },
    // },
    // BTC: {
        BTC_GATEMHCCKCY67ZUCKTROYN24ZYT5GK4EQZ65JJLDHKHRUZI3EUEKMTCH: {
            code: 'BTC', 
            issuer: 'GATEMHCCKCY67ZUCKTROYN24ZYT5GK4EQZ65JJLDHKHRUZI3EUEKMTCH',
            domain: 'naobtc.com'
        },
        BTC_GAUTUYY2THLF7SGITDFMXJVYH3LHDSMGEAKSBU267M2K7A3W543CKUEF: {
            code: 'BTC', 
            issuer: 'GAUTUYY2THLF7SGITDFMXJVYH3LHDSMGEAKSBU267M2K7A3W543CKUEF',
            domain: 'apay.io'
        },
    // },
    KIN: {
        code: 'KIN', 
        issuer: 'GBDEVU63Y6NTHJQQZIKVTC23NWLQVP3WJ2RI2OTSJTNYOIGICST6DUXR',
        domain: 'apay.io'
    },
    LTC: {
        code: 'LTC', 
        issuer: 'GC5LOR3BK6KIOK7GKAUD5EGHQCMFOGHJTC7I3ELB66PTDFXORC2VM5LP',
        domain: 'apay.io'
    },
    XRP: {
        code: 'XRP', 
        issuer: 'GA7FCCMTTSUIC37PODEL6EOOSPDRILP6OQI5FWCWDDVDBLJV72W6RINZ',
        domain: 'vcbear.net'
    },
    BCH: {
        code: 'BCH', 
        issuer: 'GAEGOS7I6TYZSVHPSN76RSPIVL3SELATXZWLFO4DIAFYJF7MPAHFE7H4',
        domain: 'apay.io'
    },
    TARI: {
        code: 'TARI', 
        issuer: 'GD7UVDDJHJYKUXB4SJFIC6VJDQ4YADQCMRN3KLHJFV4H6NIUAEREVCO7',
        domain: 'cryptotari.io'
    },
    ZRX: {
        code: 'ZRX', 
        issuer: 'GBDEVU63Y6NTHJQQZIKVTC23NWLQVP3WJ2RI2OTSJTNYOIGICST6DUXR',
        domain: 'apay.io'
    }
};


module.exports = ST_TickerIDs;