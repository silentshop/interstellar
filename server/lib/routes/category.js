"use strict";
/** Firesbase */
require("firebase");
const categoryRouter = require("express").Router();
categoryRouter.get("/", (req, res) => {
    req.db
        .collection("categories")
        .get()
        .then((collectionSnapshot) => {
            const docs = collectionSnapshot.docs.map((documentSnapshot) => documentSnapshot.data());
            res.status(res.statusCode).send(docs);
        })
        .catch((err) => {
            res.status(res.statusCode).json({ error: err.toString() });
        });
});

categoryRouter.post("/", (req, res) => {
    console.log("post category");
    const nextID = req.db.collection("categories").doc().id;
    const productCategory = req.body;
    productCategory.id = nextID;
    console.log("nextID " + nextID)    
    req.db
        .collection("categories")
        .add(productCategory)
        .then(documentSnapshot => {
            console.log(documentSnapshot.data())
            console.log(documentSnapshot)
            console.log("documentSnapshot.data()")
            res.status(res.statusCode).send(documentSnapshot.data());
        })
        .catch((err) => {
            res.status(res.statusCode).send(err);
        });
});

categoryRouter.get("/:category", (req, res) => {
    console.log("get categoried products");
    const category = req.params["category"];
    req.db
        .collection("products-categories")
        .doc(category)
        .collection("products")
        .get()
        .then((collectionSnapshot) => {
            console.log("get categoried products on server");
            const docs = collectionSnapshot.docs.map((documentSnapshot) => documentSnapshot.data());
            res.status(res.statusCode).send(docs);
        })
        .catch((err) => {
            res.status(res.statusCode).json({ error: err.toString() });
        });
});

module.exports = categoryRouter;
