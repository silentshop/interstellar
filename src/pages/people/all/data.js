// @flow

export type product = {
    name: string,
    description: string,
    numberOfSales: number,
    rating: number,
    id: string
}

export const people: Array<product> = [
    {
        id: "1",
        name: "Person 1",
        description: "This is a product.",
        numberOfSales: 10,
        rating: 4
    },
    {
        id: "2",
        name: "Person 2",
        description: "This is a product.",
        numberOfSales: 20,
        rating: 4.5
    },
    {
        id: "3",
        name: "Person 3",
        description: "This is a product.",
        numberOfSales: 30,
        rating: 3
    },
    {
        id: "4",
        name: "Person 4",
        description: "This is a product.",
        numberOfSales: 40,
        rating: 2.5
    },
    {
        id: "5",
        name: "Person 5",
        description: "This is a product.",
        numberOfSales: 50,
        rating: 4.5
    },
];

