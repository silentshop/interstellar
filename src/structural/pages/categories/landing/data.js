// @flow

export type product = {
    name: string,
    price: number,
    description: string
}

export const products: Array<product> = [
    {
        name: "Product 1",
        price: 20,
        description: "This is a product."
    },
    {
        name: "Product 2",
        price: 10,
        description: "This is a product."
    },
    {
        name: "Product 3",
        price: 2022,
        description: "This is a product."
    },
    {
        name: "Product 4",
        price: 25,
        description: "This is a product."
    },
    {
        name: "Product 5",
        price: 2,
        description: "This is a product."
    },
];

