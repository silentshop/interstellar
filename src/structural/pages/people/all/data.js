// @flow

export type product = {
    name: string,
    description: string,
    numberOfSales: number,
    rating: number
}

export const people: Array<product> = [
    {
        name: "Person 1",
        description: "This is a product.",
        numberOfSales: 10,
        rating: 4
    },
    {
        name: "Person 2",
        description: "This is a product.",
        numberOfSales: 20,
        rating: 4.5
    },
    {
        name: "Person 3",
        description: "This is a product.",
        numberOfSales: 30,
        rating: 3
    },
    {
        name: "Person 4",
        description: "This is a product.",
        numberOfSales: 40,
        rating: 2.5
    },
    {
        name: "Person 5",
        description: "This is a product.",
        numberOfSales: 50,
        rating: 4.5
    },
];

