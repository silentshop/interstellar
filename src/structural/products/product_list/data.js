// @flow

export type product = {
    name: string,
    price: number,
    description: string,
    rating: number
}

export const products: Array<product> = [
    {
        name: "Product 1",
        price: 20,
        description: "This is a product.",
        rating: 2
    },
    {
        name: "Product 2",
        price: 10,
        description: "This is a product.",
        rating: 5
    },
    {
        name: "Product 3",
        price: 2022,
        description: "This is a product.",
        rating: 4
    },
    {
        name: "Product 4",
        price: 25,
        description: "This is a product.",
        rating: 4.5
    },
    {
        name: "Product 5",
        price: 2,
        description: "This is a product.",
        rating: 3.5
    },
];

