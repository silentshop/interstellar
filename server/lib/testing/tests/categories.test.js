import * as CategoryService from '../../services/category.service'
import { categories, categoryIDs } from "../test-data/category.data";

/** test create new category */
test('Category-Service: createNewCategory(newCategory)', () => {
    CategoryService.createNewCategory = jest.fn().mockImplementation(() => {
        return new Promise((resolve, reject) => {
            resolve({
                docID: "some-new-docID"
            })
        })
    });
    CategoryService.createNewCategory("some new category")
        .then(response => 
            expect(response).toEqual({docID: "some-new-docID"})
        );
});

/** test serialize categories */
expect.addSnapshotSerializer({
    test:(val) => val.id && val.category && val.description && val.imageURL,
    print:(val) => 
     `id: ${val.id}
      category: ${val.category}
      description: ${val.description}
      imageURL: ${val.imageURL}`
})

/** test get sellers categories */
test('Category-Service: getAllCategories()', () => {
    CategoryService.getAllCategories = jest.fn().mockImplementation(() => {
        return new Promise((resolve, reject) => {
            resolve(categories)
        })
    });
    CategoryService.getAllCategories()
        .then(testCategories => 
            expect(testCategories).toMatchSnapshot()
        );
});

/** test individual categories */
categoryIDs.map(categoryID => {
    it(`Category-Service: getCategoryByID(${categoryID})`, () => {
        CategoryService.getCategoryByID = jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
                const thisCategory = categories.find(cat => cat.id === categoryID);
                resolve(thisCategory)
            })
        });
        CategoryService.getCategoryByID(categoryID)
            .then(category => {
                expect(category).toMatchSnapshot()
                expect(category).toHaveProperty("id", categoryID);
                expect(category).toEqual(
                    expect.objectContaining({
                        id: expect.any(String),
                        category: expect.any(String),
                        description: expect.any(String),
                        imageURL: expect.any(String)
                    })
                );
            });
    });
});
