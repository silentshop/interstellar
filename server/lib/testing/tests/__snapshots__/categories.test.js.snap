// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category-Service: getAllCategories() 1`] = `
Array [
  id: 7UR9yft57xqf7xNzwFPE
      category: Sports & Outdoors
      description: For your next adventure.
      imageURL: https://images.pexels.com/photos/364308/pexels-photo-364308.jpeg?cs=srgb&dl=ball-fun-game-364308.jpg&fm=jpg,
  id: 8pgPUTU6a3TvoqlmNbXk
      category: Electronics
      description: For work or play.
      imageURL: https://images.pexels.com/photos/39284/macbook-apple-imac-computer-39284.jpeg?cs=srgb&dl=apple-business-computer-39284.jpg&fm=jpg,
  id: G972IkhxOMmeeBmSIeVU
      category: Movies, Music & Games
      description: For time with friends.
      imageURL: https://images.pexels.com/photos/33129/popcorn-movie-party-entertainment.jpg?cs=srgb&dl=cinema-food-movie-theater-33129.jpg&fm=jpg,
  id: IBOtrlgVsbpBAyh9WB0c
      category: Food & Grocery
      description: For when you need fuel.
      imageURL: https://images.pexels.com/photos/459469/pexels-photo-459469.jpeg?cs=srgb&dl=basil-delicious-food-459469.jpg&fm=jpg,
  id: Q54jqKDasMhgkKUMM2ec
      category: Beauty & Health
      description: For you.
      imageURL: https://images.pexels.com/photos/6148/brush-makeup-make-up-brushes.jpg?cs=srgb&dl=brushes-make-up-makeup-6148.jpg&fm=jpg,
  id: TY399lFhZWeaob0DbUX8
      category: Home, Garden, Pets & Tools
      description: For your home.
      imageURL: https://images.pexels.com/photos/597329/pexels-photo-597329.jpeg?cs=srgb&dl=art-materials-arts-and-crafts-bloom-597329.jpg&fm=jpg,
  id: jttEhnNzRklYeGJG1lFG
      category: Books & Audio
      description: For your next rainy day.
      imageURL: https://images.pexels.com/photos/159711/books-bookstore-book-reading-159711.jpeg?cs=srgb&dl=bookcase-books-bookshelves-159711.jpg&fm=jpg,
  id: pGnm5AN2RnYpuVhwkko6
      category: Clothing, Shoes & Jewelry
      description: For that next night out.
      imageURL: https://images.pexels.com/photos/322207/pexels-photo-322207.jpeg?cs=srgb&dl=accessories-accessory-bag-322207.jpg&fm=jpg,
]
`;

exports[`Category-Service: getCategoryByID(7UR9yft57xqf7xNzwFPE) 1`] = `
id: 7UR9yft57xqf7xNzwFPE
      category: Sports & Outdoors
      description: For your next adventure.
      imageURL: https://images.pexels.com/photos/364308/pexels-photo-364308.jpeg?cs=srgb&dl=ball-fun-game-364308.jpg&fm=jpg
`;

exports[`Category-Service: getCategoryByID(8pgPUTU6a3TvoqlmNbXk) 1`] = `
id: 8pgPUTU6a3TvoqlmNbXk
      category: Electronics
      description: For work or play.
      imageURL: https://images.pexels.com/photos/39284/macbook-apple-imac-computer-39284.jpeg?cs=srgb&dl=apple-business-computer-39284.jpg&fm=jpg
`;

exports[`Category-Service: getCategoryByID(G972IkhxOMmeeBmSIeVU) 1`] = `
id: G972IkhxOMmeeBmSIeVU
      category: Movies, Music & Games
      description: For time with friends.
      imageURL: https://images.pexels.com/photos/33129/popcorn-movie-party-entertainment.jpg?cs=srgb&dl=cinema-food-movie-theater-33129.jpg&fm=jpg
`;

exports[`Category-Service: getCategoryByID(IBOtrlgVsbpBAyh9WB0c) 1`] = `
id: IBOtrlgVsbpBAyh9WB0c
      category: Food & Grocery
      description: For when you need fuel.
      imageURL: https://images.pexels.com/photos/459469/pexels-photo-459469.jpeg?cs=srgb&dl=basil-delicious-food-459469.jpg&fm=jpg
`;

exports[`Category-Service: getCategoryByID(Q54jqKDasMhgkKUMM2ec) 1`] = `
id: Q54jqKDasMhgkKUMM2ec
      category: Beauty & Health
      description: For you.
      imageURL: https://images.pexels.com/photos/6148/brush-makeup-make-up-brushes.jpg?cs=srgb&dl=brushes-make-up-makeup-6148.jpg&fm=jpg
`;

exports[`Category-Service: getCategoryByID(TY399lFhZWeaob0DbUX8) 1`] = `
id: TY399lFhZWeaob0DbUX8
      category: Home, Garden, Pets & Tools
      description: For your home.
      imageURL: https://images.pexels.com/photos/597329/pexels-photo-597329.jpeg?cs=srgb&dl=art-materials-arts-and-crafts-bloom-597329.jpg&fm=jpg
`;

exports[`Category-Service: getCategoryByID(jttEhnNzRklYeGJG1lFG) 1`] = `
id: jttEhnNzRklYeGJG1lFG
      category: Books & Audio
      description: For your next rainy day.
      imageURL: https://images.pexels.com/photos/159711/books-bookstore-book-reading-159711.jpeg?cs=srgb&dl=bookcase-books-bookshelves-159711.jpg&fm=jpg
`;

exports[`Category-Service: getCategoryByID(pGnm5AN2RnYpuVhwkko6) 1`] = `
id: pGnm5AN2RnYpuVhwkko6
      category: Clothing, Shoes & Jewelry
      description: For that next night out.
      imageURL: https://images.pexels.com/photos/322207/pexels-photo-322207.jpeg?cs=srgb&dl=accessories-accessory-bag-322207.jpg&fm=jpg
`;
